ui/js/controllers/main.js

```diff
@@ -22,7 +22,59 @@ treeherderApp.controller('MainCtrl', [
 
         $rootScope.getWindowTitle = function() {
             var ufc = $scope.getUnclassifiedFailureCount($rootScope.repoName);
-            return "[" + ufc + "] " + $rootScope.repoName;
+            var params = $location.search();
+
+            // repoName is undefined for the first few title update attempts, show something sensible
+            var title = "[" + ufc + "] " + ($rootScope.repoName ? $rootScope.repoName : "Treeherder");
+
+            if (params["revision"]) {
+                var desc = getSingleRevisionTitleString();
+                var revtitle = desc[0] ? ": " + desc[0] : "";
```

Comment:
This should still just work, since undefined is falsy, it should pick the second option "" and not mess with the title. I could change the function to return [false,false] if you'd prefer for it to be more explicit.

http://localhost:8000/#/jobs?repo=try&revision=5f8c1b6e93750aa0bd86aa411ef2534d0b638f70 is a revision where that function returns false, and everything appears to be working correctly, with no errors logged to the console.